path_decrypt.go

```diff
@@ -7,14 +7,18 @@ import (
 	"context"
 	"encoding/base64"
 	"fmt"
+	"time"
 
 	"github.com/hashicorp/errwrap"
 	"github.com/hashicorp/vault/sdk/framework"
 	"github.com/hashicorp/vault/sdk/logical"
+	"github.com/patrickmn/go-cache"
 
 	kmspb "google.golang.org/genproto/googleapis/cloud/kms/v1"
 )
 
+var keysCache = cache.New(24*time.Hour, 10*time.Minute)
```

Comment:
Thoughts on making it a field in the plugin's backend for cohesion and so each instance has its own cache?
```
type backend struct {
        *framework.Backend

        keysCache *cache.Cache
        (...)
}
```